### ISSUE TITLE
Rewrite cli/sql_util.go to use database/sql interface

### ISSUE THREAD

**Issue by spencerkimball** [cockroachdb/cockroach#14964]
Since go1.8 has support for multiple result sets, `cli/sql_util.go` should be rewritten to use `database/sql` directly instead of directly accessing the `database/sql/driver`.

In addition, we should eliminate the duplicative code in `sqlConn.ExecTxn` and use `cockroach-go/crdb` instead.

**Comment by mjibson** [comment]
I do not think it possible to use `database/sql` until go 1.9. We use the `Result()` and `Tag()` [methods](https://github.com/cockroachdb/cockroach/blob/00185f3df8fc170719a87083632e820891e47264/pkg/cli/sql_util.go#L110) to fetch the tag of any exec'd statements and number of rows affected. This isn't available through the `sql.Rows` type, because it's a struct that doesn't export types, so we can't get to the underlying driver.Rows.

Go 1.9 is adding a `.Message()` method to the sql package which will allow for us to finally use `database/sql`, support for which I'm going to add to `lib/pq`.
